src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -836,6 +836,28 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
 
             return response.Data.ToDictionary(kv => kv.Key, kv => kv.Value);
         }
+
+        /// <inheritdoc/>
+        public async override ValueTask<Dictionary<string, string>> GetBulkSecretAsync(
```

Comment:
Is the return value correct here? `GetSecretAsync` returns `ValueTask<Dictionary<string, string>>`, so I would expect this to be `ValueTask<Dictionary<string, Dictionary<string, string>>>`

Reply:
This is an eye opener. They way I implemented this in **components-contrib** is returning the `map[string]string`, so for instance when using [kubernetes](https://github.com/dapr/components-contrib/blob/4d3e1be105ef2c10f64d6b80f87341244a6751e0/secretstores/kubernetes/kubernetes.go#L78) the store is only returning the underlying secret maps and it's ignoring the original secret name.

I guess all this means that we'll have to revisit both the **dapr** and **components-contrib** methods and Bulk Secrets API in order to return the expected information.

Reply:
So if I'm understanding this right - the implementation of "Bulk Secrets" currently will - flatten all of the secrets into a single map, which is why the API looks like it does.

Also based on the code you linked, it looks like it ignores duplicates when flattening.

Reply:
Did you create an issue for this?

Reply:
I'll create an issue in both the components and dapr repos to change the behavior and then be able to return `ValueTask<Dictionary<string, Dictionary<string, string>>>`  as expected here.

Agree?

Reply:
sounds good, let's start a discussion there - I don't claim to know everything, but the current design is not what I expected.

Also note, we just changed to use `Task` so when you do another round `ValueTask<>` -> `Task<>`

Reply:
Waiting for https://github.com/dapr/components-contrib/issues/590 and https://github.com/dapr/dapr/issues/2652 to be fixed.